src/circuit/tx.rs

```diff
@@ -175,14 +187,52 @@ pub fn c_transfer<C:CS, P:PoolParams<Fr=C::Fr>>(
         (&s.tx.input.1[i].p_d - c_derive_key_p_d(&s.tx.input.1[i].d.as_num(), &eta_bits, params).x).assert_zero();
     }
 
+    // Check daily turnover limit and transfer limit
+    let in_account = &s.tx.input.0;
+    let out_account = &s.tx.output.0;
+    {
+        let is_new_day = c_comp(&p.day.as_num(), &in_account.last_action_day.as_num(), DAY_SIZE_BITS);
+
+        let delta_value_is_positive = c_comp(&total_value, &p.derive_const(&Num::ZERO), TURNOVER_SIZE_BITS);
+        let delta_value_abs = total_value.clone().switch(&delta_value_is_positive, &-&total_value);
+        let tx_turnover = delta_value_abs + &out_notes_sum;
+        let turnover = tx_turnover.switch(&is_new_day, &(in_account.daily_turnover.as_num() + &tx_turnover));
+
+        // Check transfer limit
+        c_comp(&out_notes_sum, &p.transfer_limit.as_num(), BALANCE_SIZE_BITS).assert_const(&false);
+        // Check that current day >= last_action_day
+        (&is_new_day | p.day.is_eq(&in_account.last_action_day)).assert_const(&true);
+        // Check turnover limit
+        c_comp(&turnover, &p.daily_limit.as_num(), TURNOVER_SIZE_BITS).assert_const(&false);    
+        // Check output account turnover
+        out_account.daily_turnover.as_num().is_eq(&turnover).assert_const(&true);
+        // Check output account last_action_day
+        out_account.last_action_day.as_num().is_eq(&p.day.as_num()).assert_const(&true);
+    }
+
+    //assuming input_pos_index <= current_index
+    let ref input_pos_index = c_from_bits_le(s.in_proof.0.path.as_slice());
+
+    // Check in_account_hash. If account is old, last_action_day and daily_turnover must be zero
+    let (in_account_hash, nullifier) = {
+        let in_account_hash_new = in_account.hash(params);
+        let in_account_hash_old = in_account.hash_old(params);
+
+        let nullifier_new = c_nullfifier(&in_account_hash_new, &eta, input_pos_index, params);
+        let nullifier_old = c_nullfifier(&in_account_hash_old, &eta, input_pos_index, params);
+
+        let is_old_account = nullifier_old.is_eq(&p.nullifier);
+        (!&is_old_account | (in_account.last_action_day.as_num() + in_account.daily_turnover.as_num()).is_zero()).assert_const(&true);

```

Comment:
Can user make a double spend during the migration of the account, by using two nullifiers (`nullifier_new` and `nullifier_old`) in parallel? If not, what prevents such case?

Reply:
It seems that user can't do it because there is no commitment in mekrle tree that corresponds to the new nullifier. But I'll think more about it. It is a good point, thanks!

Reply:
Yeah, I thought about that too, but the merkle root is determiner either by the old or the new account hash in a single moment, but not both, so there seems to be no way to get a valid proof for both at the same time
https://github.com/zkBob/libzeropool/pull/3/files#diff-8f588a3149a512abb6d2d301fd6baa95b46c51ab894adba38c48595d1568764dR237